lib/internal/Magento/Framework/Module/ModuleList/Sorter.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * @copyright Copyright (c) 1999-2017 netz98 GmbH (http://www.netz98.de)
+ *
+ * @see PROJECT_LICENSE.txt
+ */
+
+namespace Magento\Framework\Module\ModuleList;
+
+class Sorter
+{
+    /**
+     * @var array
+     */
+    private $origList;
+
+    /**
+     * @var array
+     */
+    private $elements = [];
+
+    /**
+     * @var array
+     */
+    private $sortedModules;
+
+    /**
+     * @param array $origList
+     * @return array
+     */
+    public function sort(array $origList)
+    {
+        $this->origList = $origList;
+
+        foreach ($origList as $moduleName => $moduleInformation) {
+            $this->elements[$moduleName] = [
+                'name'          => $moduleName,
+                'setup_version' => $moduleInformation['setup_version'],
+                'sequence'      => $moduleInformation['sequence'],
+                'isProcessed'   => false,
+            ];
+        }
+
+        $this->sortedModules = [];
+
+        foreach ($this->elements as $element) {
+            $this->processElement($element, []);
+        }
+
+        return $this->sortedModules;
+    }
+
+    /**
+     * @param array $element
+     * @param array $parentElements
+     * @throws \Exception
+     */
+    protected function processElement($element, $parentElements = [])
```

Comment:
Could this be updated to private since I think it is only used inside this class?